crates/sncast/src/helpers/configuration.rs

```diff
@@ -99,4 +246,49 @@ mod tests {
             "Failed to find snfoundry.toml - not found in current nor any parent directories"
         );
     }
+
+    #[test]
+    fn parse_config_happy_case_with_profile() {
+        let tempdir = copy_config_to_tempdir("tests/data/files/correct_snfoundry.toml", None);
+        let config = parse_config(
+            &Some(String::from("profile1")),
+            &Some(Utf8PathBuf::try_from(tempdir.path().to_path_buf()).unwrap()),
+        )
+        .unwrap();
+
+        assert_eq!(config.account, String::from("user3"));
+        assert_eq!(config.rpc_url, String::from("http://127.0.0.1:5050/rpc"));
+    }
+
+    #[test]
+    fn parse_config_happy_case_default_profile() {
+        let tempdir = copy_config_to_tempdir("tests/data/files/correct_snfoundry.toml", None);
+        let config = parse_config(
+            &None,
+            &Some(Utf8PathBuf::try_from(tempdir.path().to_path_buf()).unwrap()),
+        )
+        .unwrap();
+        assert_eq!(config.account, String::from("user1"));
+        assert_eq!(config.rpc_url, String::from("http://127.0.0.1:5055/rpc"));
+    }
+
+    #[test]
+    fn parse_config_not_found() {
+        let tempdir = TempDir::new().expect("Failed to create a temporary directory");
+        let config = parse_config(
+            &None,
+            &Some(Utf8PathBuf::try_from(tempdir.path().to_path_buf()).unwrap()),
+        )
+        .unwrap();
+
+        assert_eq!(config.account, String::new());
+        assert_eq!(config.rpc_url, String::new());
+    }
+
+    #[test]
+    fn test_config_defaults() {
```

Comment:
nit: Maybe also add a test for `parse_config` when the path is `None`?

Reply:
This is basically impossible to test in unit tests :( I have experimented with sealed tests etc, but it all went down to the fact that we would need to set cwd for this test, and since the test env for all tests is the same, the other tests were executed in the new dir as well, and this resulted in many errors.

I do, however, test exactly this scenario in e2e tests, where we are able to set per-snapbox working directory, so it is generally covered :+1: